src/coreclr/jit/compiler.h

```diff
@@ -1577,6 +1577,53 @@ inline constexpr FlowGraphUpdates operator&(FlowGraphUpdates a, FlowGraphUpdates
     return (FlowGraphUpdates)((unsigned int)a & (unsigned int)b);
 }
 
+// Profile checking options
+//
+// clang-format off
+enum class ProfileChecks : unsigned int
+{
+    CHECK_NONE          = 0,
+    CHECK_CLASSIC       = 1 << 0, // check "classic" jit weights
+    CHECK_LIKELY        = 1 << 1, // check likelihood based weights
+    RAISE_ASSERT        = 1 << 2, // assert on check failure
+    CHECK_ALL_BLOCKS    = 1 << 3, // check blocks even if bbHasProfileWeight is false
+};
+
+inline constexpr ProfileChecks operator ~(ProfileChecks a)
+{
+    return (ProfileChecks)(~(unsigned int)a);
```

Comment:
I wonder if we should have a `CHECK_ALL` that contains all the legal bits (currently 0xF) and have this be:
```suggestion
    return (ProfileChecks)(CHECK_ALL & ~(unsigned int)a);
```
to prevent returning bits that are not "legal" `ProfileChecks` bits.

Could validate/cap the value of `JitConfig.JitProfileChecks()` this way, too.

Reply:
This is all boilerplate for our common `enum class` approach, so we'd probably want to try and handle them all this way. 

Since `enum class` is (somewhat) type safe seems like we could just mask when converting from `int` or other similar types?

Reply:
Sure, I hadn't noticed that other similar cases had also defined `operator~()` without this. A minor thing, to be sure.